source/hpack.c

```diff
@@ -1236,9 +1241,10 @@ int aws_hpack_decode(
                 if (!size_complete) {
                     break;
                 }
-
-                if (*size64 > SIZE_MAX) {
-                    HPACK_LOG(ERROR, context, "Dynamic table update size is absurdly large");
+                /* The new maximum size MUST be lower than or equal to the limit determined by the protocol using HPACK.
+                 * A value that exceeds this limit MUST be treated as a decoding error. */
+                if (*size64 > context->dynamic_table.protocol_max_size_setting) {
+                    HPACK_LOG(ERROR, context, "Dynamic table update size is larger than the protocal setting");
```

Comment:
I'm not really sure this check is correct
maybe we should just remove all the changes to hpack for now and tackle it in another PR